File: src/main/java/ting/leetcode/java/arrays101/CorporateFlightBookings.java
package ting.leetcode.java.arrays101;

public class CorporateFlightBookings {
    public int[] corpFlightBookings(int[][] bookings, int n) {
        int[] bookedSeat = new int[n];
        for (int[] booking : bookings) {
            increase(booking[0], booking[1], booking[2], bookedSeat);
        }

        for (int i = 1; i < bookedSeat.length; i++) {
            bookedSeat[i] = bookedSeat[i - 1] + bookedSeat[i];
        }
        return bookedSeat;
    }

    public void increase(int from, int to, int count, int[] arr) {
        arr[from - 1] += count;
        if (to - 1 < arr.length - 1) {
            arr[to] -= count;
        }
    }

    public static void main(String[] args) {
        int[][] array = {
                {1, 2, 10},
                {2, 3, 20},
                {2, 5, 25}
        };
        new CorporateFlightBookings().corpFlightBookings(array, 5);
    }
}

File: src/main/kotlin/ting/leetcode/arrays101/CorporateFlightBookings.kt
package ting.leetcode.arrays101

class CorporateFlightBookings {
    fun corpFlightBookings(bookings: Array<IntArray>, n: Int): IntArray {
        val bookedSeat = IntArray(n)
        for (booking in bookings) {
            increase(booking, bookedSeat)
        }

        for (i in 1..bookedSeat.lastIndex) {
            bookedSeat[i] += bookedSeat[i - 1]
        }
        return bookedSeat
    }

    private fun increase(booking: IntArray, array: IntArray) {
        val from = booking.first() - 1
        val to = booking[1] - 1
        val count = booking.last()
        array[from] += count
        if (to < array.lastIndex)
            array[to] -= count
    }
}
